packages/liveblocks-react/src/room.tsx

```diff
@@ -1225,39 +1272,27 @@ export function createRoomContext<
 
     React.useEffect(
       () => {
-        if (state.isLoading) return;
+        if (state.isLoading === true) return;
 
-        const hash = window.location.hash;
-        const commentId = hash.slice(1);
-
-        if (!commentId.startsWith("cm_")) return;
-
-        const comments = state.threads.flatMap((thread) => thread.comments);
-        const isCommentInThreads = comments.some(
-          (comment) => comment.id === commentId
-        );
-
-        // If the comment is not in the threads for this hook, we do not scroll to it
-        if (!isCommentInThreads) return;
-
-        const comment = document.getElementById(commentId);
-        if (comment === null) return;
-
-        comment.scrollIntoView();
+        handleScrollToCommentOnLoad(state.isLoading, scrollOnLoad, state);
       },
       // eslint-disable-next-line react-hooks/exhaustive-deps -- We only want to run this effect once
-      [state.isLoading]
+      [state.isLoading, scrollOnLoad]
```

Comment:
Should we "freeze" `scrollOnLoad` into a ref and not trigger the effect based on it? I don't think we want to support switching from `false` to `true` later in the lifecycle and then triggering the behavior.